extends Control

func _ready():
	Input.set_mouse_mode(Input.MOUSE_MODE_VISIBLE)
	print("Men煤 Principal cargado")

func _input(event):
	if event is InputEventKey and event.pressed:
		if event.keycode == KEY_ESCAPE:
			get_tree().quit()

func _on_boton_historia_pressed() -> void:
	print("隆Bot贸n HISTORIA presionado!")
	
	# Inicializar modo historia
	Juego.iniciar_modo_historia()
	
	# Ir a la pantalla de transici贸n
	get_tree().change_scene_to_file("res://EscenasGenerales/Cinematicas/CinematicaGatoDurmiendo.tscn")

func _on_boton_infinito_pressed() -> void:
	print("隆Bot贸n MODO INFINITO presionado!")
	
	# Inicializar modo infinito
	Juego.iniciar_modo_infinito()
	
	# Ir a la pantalla de transici贸n
	get_tree().change_scene_to_file("res://EscenasGenerales/EscenaVidasNumeroMicrojuego/MicroInicioContadorVidas.tscn")



func _on_boton_opciones_pressed() -> void:

	print("隆Bot贸n OPCIONES presionado!")

	#  NUEVO: Cargar el men煤 de opciones

	get_tree().change_scene_to_file("res://EscenasGenerales/Menus/MenuOpciones.tscn")

func _on_boton_salir_pressed() -> void:
	print("隆Bot贸n SALIR presionado!")
	get_tree().quit()
	

func _on_boton_multijugador_pressed() -> void:
	print("隆Bot贸n MULTIJUGADOR presionado!")
	get_tree().change_scene_to_file("res://EscenasGenerales/Menus/Online/user_login.tscn")
